### DISCUSSION TITLE
When I edit vue code, there is always an error notification like this.

### DISCUSSION THREAD

**Discussion by iamxiaojianzheng** [LazyVim/LazyVim#3461] [Q&A]
I tried to search for related questions and only saw a similar situation in this link, but I didn't find the plug-in in LazyVim. 
So does anyone know what's going on?

https://github.com/hadronized/hop.nvim/issues/361

```
Error in decoration provider vim_lsp_inlayhint.win:
Error executing lua: ...im/current/share/nvim/runtime/lua/vim/lsp/inlay_hint.lua:359: Invalid 'col': out of range
stack traceback:
	[C]: in function 'nvim_buf_set_extmark'
	...im/current/share/nvim/runtime/lua/vim/lsp/inlay_hint.lua:359: in function <...im/current/share/nvim/runtime/lua/vim/lsp/inlay_hint.lua:320>
	[C]: in function 'nvim_exec2'
	vim/_editor.lua:431: in function 'cmd'
	...l/nvim-data/lazy/nvim-notify/lua/notify/service/init.lua:63: in function 'push'
	...Local/nvim-data/lazy/nvim-notify/lua/notify/instance.lua:77: in function <...Local/nvim-data/lazy/nvim-notify/lua/notify/instance.lua:41>
	...m-data/lazy/noice.nvim/lua/noice/view/backend/notify.lua:169: in function '_notify'
	...m-data/lazy/noice.nvim/lua/noice/view/backend/notify.lua:199: in function 'show'
	.../Local/nvim-data/lazy/noice.nvim/lua/noice/view/init.lua:155: in function <.../Local/nvim-data/lazy/noice.nvim/lua/noice/view/init.lua:153>
	[C]: in function 'xpcall'
	.../Local/nvim
```

**Comment by unr** [comment]
This has been eating me alive -- I experience the same thing.

It seems to occur if there is inlay hints for typescript, and it goes outside of the window width?

**Comment by mirsella** [comment]
have you enabled the vue extra ?
doesn't happen to me, make sure to test vanilla lazyvim to see if it's your config of lazyvim

**Reply by iamxiaojianzheng** [reply]
> have you enabled the vue extra ?你启用了 Vue extra 吗？ doesn't happen to me, make sure to test vanilla lazyvim to see if it's your config of lazyvim不会发生在我身上，请务必测试 vanilla lazyvim，看看它是否是您的 lazyvim 配置

I has enabled vue extra. 

I found others in discussions who had raised the same problem and offered temporary solutions.

https://github.com/LazyVim/LazyVim/discussions/3249

```
  {
    "neovim/nvim-lspconfig",
    -- Enable this to enable the builtin LSP inlay hints on Neovim >= 0.10.0
    -- Be aware that you also will need to properly configure your LSP server to
    -- provide the inlay hints.
    inlay_hints = {
      enabled = true,
      exclude = { "vue" }, -- filetypes for which you don't want to enable inlay hints
    },
  },
```

**Reply by mirsella** [reply]
there was recent change on the volar lsp and lazyvim config to fix it.
might be a "old" project, or vue-language-server not up to date ? what's volar version?

**Reply by iamxiaojianzheng** [reply]
Volar version is 2.0.21, I always keep the latest version.

I used it to develop in vue2，Is it because of vue2?

**Reply by mirsella** [reply]
oh, I'm only guessing but it probably is.
you can try a brand new lastest vue projet just to test if inlay hints works

**Comment by mirsella** [comment]
so what was the solution ?

**Reply by iamxiaojianzheng** [reply]
😣 I haven't found a fundamental solution yet, use the following configuration for a temporary fix.

```lua
{
    "neovim/nvim-lspconfig",
    -- Enable this to enable the builtin LSP inlay hints on Neovim >= 0.10.0
    -- Be aware that you also will need to properly configure your LSP server to
    -- provide the inlay hints.
    inlay_hints = {
      enabled = true,
      exclude = { "vue" }, -- filetypes for which you don't want to enable inlay hints
    },
  },
```

**Reply by mirsella** [reply]
Alright, good luck and keep us updated if you try vue3 !

**Comment by unr** [comment]
I've noticed the issue also occurs on .ts files, not just .vue

Seems like a core inlay_hint error, not LazyVim specific.


If it starts to pop up for me, I just temporarily turn off hints with `<leader>uh`

**Reply by mirsella** [reply]
is volar also running on .ts ? otherwise, it might be tsserver showing the errors

**Reply by unr** [reply]
It has nothing to do with volar specifically, it is the same error as shown in the first comment in this issue.

I think it's just inlay_hint itself being upset that a bunch of hinting is going off the bounds of the window.

**Reply by mirsella** [reply]
Isn't what's triggering the out of bound inlay hint the LSP like tsserver or volar sending wrong coordinates ?
i don't really know just making guess

**Comment by iamxiaojianzheng** [comment]
It seems that I have found the condition that triggers this exception.

When volar, vtsls, and inlay_hints are all enabled at the same time, the exception occurs. If any of them is disabled, the exception will not occur.

The root cause, I suspect, is that the inlay_hints cannot properly handle the situation where two LSPs, volar and vtsls, are both processing the vue file simultaneously.

Is there any way to specify that only vtsls is activated by inlay_hints, while volar is ignored. volar doesn't feature inlay_hints, according to my tests.

There is a [related problem](https://www.reddit.com/r/neovim/comments/1axlen1/inlay_hints_invalid_col_out_of_range/?rdt=44076)

**Reply by folke** [reply]
If volar doesn't support inlay_hints then it wouldn't cause issues?

**Reply by iamxiaojianzheng** [reply]
Well, I just tested it back and forth between the vue2 and vue3 projects. It feels like this problem is unique to vue2.

**Reply by mirsella** [reply]
well
> oh, I'm only guessing but it probably is (vue2).

> Alright, good luck and keep us updated if you try vue3 !

I don't think there is much you can do, maybe if volar support special related config for vue2 ?
otherwise detect if the project is vue2 and change your lazy.nvim config to disable inlay hints.

it's probably a problem internal to volar (volar changed it's config method recently too) and i dunno if vue2 will get any support.

good luck !

**Comment by iamxiaojianzheng** [comment]
I think this problem can end here, after a period of tossing and turning, finally found that this problem is caused by vue2 and volar.

on Vue2 project
1. Problems occur when volar+vtsls+inlay_hints is enabled
2. If only vtsls+inlay_hints is enabled, no problems will occur

on Vue3 project，this problem did not arise at all.

**Reply by mirsella** [reply]
If you want, you can modify the lspconfig spec and disable inlay_hints if the project is vue2 (probably check with the package.json)
